app.py

```diff
@@ -7,119 +7,129 @@
 def home():
-	if request.method == 'POST':
-		session.permanent = True
-		username = request.form['username']
-		password = request.form['password']
-		userid = 1
-		user  = User.query.filter_by(username=username).first()
-		if user and bcrypt.check_password_hash(user.password,password):	
-			session['user_id'] = userid
-			return redirect(url_for('dashboard'))
-		flash('The password is incorrect!')
-		return redirect(url_for('home'))
-		""" else:
-		if 'user_id' in session:
-			return redirect(url_for('dashboard'))"""			
-	
-
-	return render_template("login.html")
-
-
-@app.route('/dashboard', methods=['GET','POST'])
+    if request.method == 'POST':
+        session.permanent = True
+        username = request.form['username']
+        password = request.form['password']
+        userid = 1
+        user = User.query.filter_by(username=username).first()
+        if user and bcrypt.check_password_hash(user.password, password):
+            session['user_id'] = userid
+            session['username'] = username
+            return redirect(url_for('dashboard'))
+        flash('The password is incorrect!')
+        return redirect(url_for('home'))
+
+    return render_template("login.html")
+
+
+@app.route('/dashboard', methods=['GET', 'POST'])
 def dashboard():
-	start_date = '2022-07-01'
-	end_date = 'today'
-	if 'user_id' in session:
-		user_id = session['user_id']
-		username = 'karen'
-		if request.method =='POST':
-			start_date = request.form['start_date']
-			end_date = request.form['end_date']
-			
-			if start_date > end_date:
-				flash('Please enter correct dates')
-				start_date = '2022-07-01'
-				end_date = 'today'
-			if start_date=='' or end_date == '' :
-				flash('Please enter a start date and end date')
-				start_date = '2022-07-01'
-				end_date = 'today'
-
-			if start_date < '2022-07-11':
-				flash('please enter a start date after July 11th')
-				start_date = '2022-07-01'
-				end_date = 'today'
-
-			if start_date > str(date.today()):
-				flash('You cannot enter a date in the future')
-				start_date = '2022-07-01'
-				end_date = 'today'
-
-			
-			
-
-		return render_template("dashboard.html", username=username, terms_dict= ga.get_terms(), analytics_dict=ga.get_analytics_date(start_date,end_date))
-	else:
-		return redirect(url_for('home')) 
-	return render_template('dashboard.html', username=username)
-
-@app.route('/search_terms', methods=['GET','POST'])	
+    global start_date
+    end_date = 'today'
+    if 'user_id' in session:
+        user_id = session['user_id']
+        username = session['username']
+        if request.method == 'POST':
+            start_date = request.form['start_date']
+            end_date = request.form['end_date']
+
+            correct_date = date_check(start_date, end_date)
+            start_date = correct_date['startdate']
+            end_date = correct_date['enddate']
+            if correct_date['message'] != '':
+                start_date = '2022-07-01'
```

Comment:
so here you could do `start_date = START_DATE`, for example, and you'd know to look for a constant named that way.